src/Common/Commands.Common/CustomAttributes/BreakingChangeAttributeHelper.cs

```diff
@@ -0,0 +1,162 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+#undef DEBUG
+
+using Microsoft.WindowsAzure.Commands.Common.Properties;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.WindowsAzure.Commands.Common.CustomAttributes
+{
+    public class BreakingChangeAttributeHelper
+    {
+        public const string SUPPRESS_ERROR_OR_WARNING_MESSAGE_ENV_VARIABLE_NAME = "SuppressAzurePowerShellBreakingChangeWarnings";
+
+        public const string BREAKING_CHANGE_DOCUMENTATION_HEADER = @"<!--
+    Please leave this section at the top of the breaking change documentation.
+
+    New breaking changes should go under the section titled 'Current Breaking Changes', and should adhere to the following format:
+    ## Current Breaking Changes
+
+    The following cmdlets were affected this release:
+
+    **Cmdlet 1**
+    - Description of what has changed
+
+    ```powershell
+    # Old
+    # Sample of how the cmdlet was previously called
+
+    # New
+    # Sample of how the cmdlet should now be called
+    ```
+
+    ## Release X.0.0
+
+    The following cmdlets were affected this release:
+
+    **Cmdlet 1**
+    - Description of what has changed
+
+    ```powershell
+
+    # Old\
+    # Sample of how the cmdlet was previously called
+
+    # New
+    # Sample of how the cmdlet should now be called
+    ```
+
+    Note: the above sections follow the template found in the link below:
+
+    https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes/breaking-change-template.md
+    -->";
+
+        /**
+         * This function takes in a Type (expected to be a derevative of the AzurePSCmdlet)
+         * And reads all the deprecation attributes attached to it
+         * Prints a message on the cmdline For each of the attribute found
+         * 
+         * the boundParameterNames is a list of parameters bound to the cmdlet at runtime, 
+         * We only process the Parameter beaking change attributes attached only params listed in this list (if present)
+         * */
+        public static void ProcessCustomAttributesAtRuntime(Type type, InvocationInfo invocationInfo)
+        {
+            bool emitWarningOrError = true;
+
+            try
+            {
+                emitWarningOrError = bool.Parse(System.Environment.GetEnvironmentVariable(SUPPRESS_ERROR_OR_WARNING_MESSAGE_ENV_VARIABLE_NAME));
+            }
+            catch (Exception)
+            {
+                //no action
+            }
+
+            if (!emitWarningOrError)
```

Comment:
@praries880 since the environment variable is `SuppressAzurePowerShellBreakingChangeWarnings`, wouldn't we want to skip if it is set to `true`? The `emitWarningOrError` should probably be renamed to `suppressWarningOrError` to accurately reflect the value we are getting